lib/service/computeplan.go

```diff
@@ -15,8 +15,10 @@ type ComputePlanAPI interface {
 	QueryPlans(p *common.Pagination, filter *asset.PlanQueryFilter) ([]*asset.ComputePlan, common.PaginationToken, error)
 	ApplyPlanAction(key string, action asset.ComputePlanAction, requester string) error
 	UpdatePlan(computePlan *asset.UpdateComputePlanParam, requester string) error
+	FailPlan(plan *asset.ComputePlan) error
```

Comment:
Accepting a CP means the caller has to fetch the plan beforehand: is there a rationale for accepting the plan rather than its key?
Also, since this should be called from task update, there is no need to make to method public (otherwise we would have to check that the requester has the right to fail the CP).
```suggestion
	failPlan(planKey string) error
```

Reply:
Good points, thanks for raising them! There was no particular reason for doing that. I implemented the suggested changes in 6a5a605dd77062fa0c7fe1311e69bbb3e708c789.